### PR TITLE
Bgp graceful restart1

### PR DESCRIPTION
**Pull request by cprabha**
Hi,

Created new pull request to address review comments from PR 296.
https://github.com/openconfig/featureprofiles/pull/296
>>>>>>>>>>>>>>

I have added code for
1.) configuring network instance type and router id.
2.) AclCIDR which was given in script was using ip prefix instead of subnet. I have updated this.
3.) Peer group was not configured in bgp, updated this.

Issue ref: https://github.com/openconfig/featureprofiles/issues/274

Thanks,
Prabha

### CODE REVIEW

**Comment by cprabha** [comment]
@Ankur19 , please could you review this PR, I have closed earlier PR as i was facing issue with conflicts.

File: feature/bgp/gracefulrestart/ate_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go [hunk 1]
```diff
@@ -374,10 +382,19 @@ func TestTrafficWithGracefulRestartSpeaker(t *testing.T) {
 		t.Log("Start DUT interface Config")
 		configureDUT(t, dut)
 	})
+
+	// Configure Network Instance on the DUT
```
Comment (Ankur19) [inline feature/bgp/gracefulrestart/ate_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go:386]: Can you move this block to the **configureDUT()** function

File: feature/bgp/gracefulrestart/ate_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go [hunk 2]
```diff
@@ -374,10 +382,19 @@ func TestTrafficWithGracefulRestartSpeaker(t *testing.T) {
 		t.Log("Start DUT interface Config")
 		configureDUT(t, dut)
```
Comment (Ankur19) [inline feature/bgp/gracefulrestart/ate_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go:383]: **configureDUT()** function configures DUT related objects. So we can move the NetworkInstance configuration to this function.

**Review by Ankur19 — CHANGES_REQUESTED** [review]

File: feature/bgp/gracefulrestart/ate_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go (hunk 1 above)
Comment (cprabha) [reply feature/bgp/gracefulrestart/ate_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go]: Done, thank you

File: feature/bgp/gracefulrestart/ate_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go (hunk 2 above)
Comment (cprabha) [reply feature/bgp/gracefulrestart/ate_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go:388]: Moved NI configurations under configureDUT(). Thank you

File: feature/bgp/gracefulrestart/ate_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go [hunk 3]
```diff
@@ -102,14 +105,19 @@ var (
...
 func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
 	dc := dut.Config()
 	i1 := dutSrc.NewInterface(dut.Port(t, "port1").Name())
 	dc.Interface(i1.GetName()).Replace(t, i1)
 
 	i2 := dutDst.NewInterface(dut.Port(t, "port2").Name())
 	dc.Interface(i2.GetName()).Replace(t, i2)
+
+	t.Log("Configure/update Network Instance")
+	dutConfNIPath := dut.Config().NetworkInstance(*deviations.DefaultNetworkInstance)
```
Comment (Ankur19) [inline feature/bgp/gracefulrestart/ate_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go:118]: ```dutConfNIPath := dc.NetworkInstance(*deviations.DefaultNetworkInstance)```

File: feature/bgp/gracefulrestart/ate_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go [hunk 4]
```diff
@@ -69,6 +70,8 @@ const (
 	plenIPv4                 = 30
 	plenIPv6                 = 126
 	bgpPort                  = 179
+	peerGrpName              = "BGP-PEER-GROUP"
+	ateCIDR                  = "192.0.2.6/32"
```
Comment (Ankur19) [inline feature/bgp/gracefulrestart/ate_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go:74]: Can you rename this to **ateDstCIDR**

**Review by Ankur19 — CHANGES_REQUESTED** [review]

File: feature/bgp/gracefulrestart/ate_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go (hunk 3 above)
Comment (cprabha) [reply feature/bgp/gracefulrestart/ate_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go]: updated, thank you

File: feature/bgp/gracefulrestart/ate_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go (hunk 4 above)
Comment (cprabha) [reply feature/bgp/gracefulrestart/ate_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go]: updated , thank you

**Review by Ankur19 — APPROVED** [review]
LGTM!

**Comment by cprabha** [comment]
Thank you.
-Prabha



Juniper Business Use Only

**Pull request merged**